### PR TITLE
Add support for Vercel Authentication + Password Protection

### LINKED ISSUE #86: Allow setting SSO and password protection
There's no way to set up SSO or password protection at this moment, which would be quite helpful because that's something I have to change manually.

Is there any chance you can add it?

image

image

### PR DESCRIPTION
**Pull request by dglsparsons**
Closes #86

### CODE REVIEW

File: client/project_update.go [hunk 1]
```diff
@@ -9,23 +9,30 @@ import (
...
+	DeploymentType string `json:"deploymentType"`
+	Password       string `json:"password"`
+}
+
 // UpdateProjectRequest defines the possible fields that can be updated within a vercel project.
 // note that the values are all pointers, with many containing `omitempty` for serialisation.
 // This is because the Vercel API behaves in the following manner:
 // - a provided field will be updated
-// - setting the field to an empty value (e.g. ”) will remove the setting for that field.
+// - setting the field to an empty value (e.g. ”") will remove the setting for that field.
```
Comment (ecklf) [inline client/project_update.go:21]: are the different double quotations intended?

Comment (dglsparsons) [reply client/project_update.go:21]: no, lol.

File: vercel/data_source_project_test.go [hunk 2]
```diff
@@ -25,6 +25,8 @@ func TestAcc_ProjectDataSource(t *testing.T) {
 					resource.TestCheckResourceAttr("data.vercel_project.test", "output_directory", ".output"),
 					resource.TestCheckResourceAttr("data.vercel_project.test", "public_source", "true"),
 					resource.TestCheckResourceAttr("data.vercel_project.test", "root_directory", "ui/src"),
+					resource.TestCheckResourceAttr("data.vercel_project.test", "vercel_authentication.protect_production", "true"),
+					resource.TestCheckResourceAttr("data.vercel_project.test", "password_protection.protect_production", "true"),
```
Comment (ecklf) [inline vercel/data_source_project_test.go:29]: I don't think we need this twice?

Comment (ecklf) [reply vercel/data_source_project_test.go:29]: Oh nvm there is one for authentication and one for password.

**Pull request merged**